x-pack/plugin/esql/compute/src/test/java/org/elasticsearch/compute/data/BasicBlockTests.java

```diff
@@ -851,9 +895,33 @@ private static <B extends Block, BB extends Block.Builder, T> void assertNullVal
         asserter.accept(randomNonNullPosition, block);
         assertTrue(block.isNull(randomNullPosition));
         assertFalse(block.isNull(randomNonNullPosition));
+        releaseAndAssertBreaker(block, block.blockFactory().breaker());
+    }
+
+    void assertZeroPositionsAndRelease(Block block) {
+        assertThat(block.getPositionCount(), is(0));
+        releaseAndAssertBreaker(block);
+    }
+
+    void assertZeroPositionsAndRelease(Vector vector) {
+        assertThat(vector.getPositionCount(), is(0));
+        releaseAndAssertBreaker(vector);
+    }
+
+    <T extends Releasable & Accountable> void releaseAndAssertBreaker(T data) {
+        releaseAndAssertBreaker(data, breaker);
+    }
+
+    static <T extends Releasable & Accountable> void releaseAndAssertBreaker(T data, CircuitBreaker breaker) {
+        Releasables.closeExpectNoException(data);
+        assertThat(breaker.getUsed(), is(0L));
```

Comment:
I don't think you are using `Accountable` on `data` right now. Had you intended to assert that the breaker is >= to the size?

Reply:
Oh yeah, I was asserting that > 0 - something was allocated, so the breaker getUsed should be positive.  I then modified the test to use a no-op breaker, so getUsed is always 0.  I think I'll revert and not use the no-op breaker, since I think that the > 0 is more useful.

Reply:
I restored the > 0 assertion